rs/bitcoin/ckbtc/minter/tests/replay_events.rs

```diff
@@ -0,0 +1,239 @@
+use candid::{CandidType, Deserialize, Principal};
+use ic_agent::Agent;
+use ic_ckbtc_minter::state::eventlog::{replay, Event};
+use ic_ckbtc_minter::state::invariants::{CheckInvariants, CheckInvariantsImpl};
+use ic_ckbtc_minter::state::{CkBtcMinterState, Network};
+use std::path::PathBuf;
+
+#[tokio::test]
+async fn should_replay_events_for_mainnet() {
+    GetEventsFile::Mainnet
+        .retrieve_and_store_events_if_env()
+        .await;
+
+    let state =
+        replay::<SkipCheckInvariantsImpl>(GetEventsFile::Mainnet.deserialize().events.into_iter())
+            .expect("Failed to replay events");
+    state
+        .check_invariants()
+        .expect("Failed to check invariants");
+
+    assert_eq!(state.btc_network, Network::Mainnet);
+    assert_eq!(state.get_total_btc_managed(), 21_723_786_340);
+}
+
+#[tokio::test]
+async fn should_replay_events_for_testnet() {
+    GetEventsFile::Testnet
+        .retrieve_and_store_events_if_env()
+        .await;
+
+    let state =
+        replay::<SkipCheckInvariantsImpl>(GetEventsFile::Testnet.deserialize().events.into_iter())
+            .expect("Failed to replay events");
+    state
+        .check_invariants()
+        .expect("Failed to check invariants");
+
+    assert_eq!(state.btc_network, Network::Testnet);
+    assert_eq!(state.get_total_btc_managed(), 16_578_205_978);
+}
+
+// This test is ignored because it takes too long to run,
+// roughly 50 minutes. It's useful to run it locally when
+// updating mainnet_events.gz or testnet_events.gz.
+// bazel test //rs/bitcoin/ckbtc/minter:ckbtc_minter_replay_events_tests --test_arg="should_replay_events_and_check_invariants" --test_arg=--ignored
+#[test]
+#[ignore]
+fn should_replay_events_and_check_invariants() {
+    for file in [GetEventsFile::Mainnet, GetEventsFile::Testnet] {
+        let events = file.deserialize();
+        println!("Replaying {} {:?} events", events.total_event_count, file);
+        let _state = replay::<CheckInvariantsImpl>(events.events.into_iter())
+            .expect("Failed to replay events");
+    }
+}
+
+// It's not clear why those events are here in the first place
+// but this test ensures that the number of such events doesn't grow.
```

Comment:
Is the long-term plan to purge them?

Reply:
I think it is a good idea to remove them, since the mainnet events are overwhelmingly "useless", not a happy situation. But on the other hand, I wonder how much of the actual space in the stable memory would be freed, because it may depend on the memory manager.

Reply:
I agree, I'm also not sure whether we could reclaim the unused space once the logs have been cleaned up, but any way the current stable memory usage is at 128 MiB so it's not that huge. I opened [XC-239](https://dfinity.atlassian.net/browse/XC-239) for this.